### PR TITLE
Saving incomplete reports' request_id in state

### PR DESCRIPTION
**Pull request by agaman**
Fixed a bug where state was saved indicating that a report was synced when it wasn't

### CODE REVIEW

File: tap_bing_ads/__init__.py [hunk 1]
```diff
@@ -43,8 +43,8 @@
 CONFIG = {}
 STATE = {}
 
-# ~30 minute job timeout
-MAX_NUM_REPORT_POLLS = 360
+# ~2 hour polling timeout
+MAX_NUM_REPORT_POLLS = 1440
```
Comment (agaman) [inline tap_bing_ads/__init__.py]: Thoughts on this?

File: tap_bing_ads/__init__.py [hunk 2]
```diff
@@ -644,20 +644,63 @@ def stream_report(stream_name, report_name, url, report_time):
...
+    download_url = await poll_report(client, report_name, start_date, end_date,
+                                     request_id)
+
+    if download_url:
+        stream_report(report_stream.stream,
+                      report_name,
+                      download_url,
+                      report_time)
+        singer.write_bookmark(STATE, state_key, 'request_id', None)
+        singer.write_bookmark(STATE, state_key, 'date', end_date.isoformat())
```
Comment (agaman) [inline tap_bing_ads/__init__.py]: Only saving the date if we have a download_url and streaming is complete.

File: tap_bing_ads/__init__.py [hunk 3]
```diff
@@ -644,20 +644,63 @@ def stream_report(stream_name, report_name, url, report_time):
...
+
+    download_url = await poll_report(client, report_name, start_date, end_date,
+                                     request_id)
+
+    if download_url:
+        stream_report(report_stream.stream,
+                      report_name,
+                      download_url,
+                      report_time)
+        singer.write_bookmark(STATE, state_key, 'request_id', None)
```
Comment (agaman) [inline tap_bing_ads/__init__.py]: Is this the right way to clear out a bookmark?

File: tap_bing_ads/__init__.py [hunk 4]
```diff
@@ -644,20 +644,63 @@ def stream_report(stream_name, report_name, url, report_time):
...
+    report_schema = get_report_schema(client, report_name)
+    singer.write_schema(report_stream.stream, report_schema, [])
+
+    report_time = arrow.get().isoformat()
+
+    request_id = get_report_request_id(client, account_id, report_stream,
+                                       report_name, start_date, end_date,
+                                       state_key)
+
+    singer.write_bookmark(STATE, state_key, 'request_id', request_id)
```
Comment (agaman) [inline tap_bing_ads/__init__.py]: Should there be a call to `singer.write_state()` after this to persist the request_id in case of failure?

Comment (KAllan357) [reply tap_bing_ads/__init__.py]: Yes - writing it as soon as possible is ideal.

Comment (agaman) [reply tap_bing_ads/__init__.py]: Updated in the latest commit.

**Comment by KAllan357** [comment]
👍

**Pull request merged**